Application/src/net/Decode.java

```diff
@@ -0,0 +1,142 @@
+package net;
+
+import javafx.application.Platform;
+import visuals.DiagramView;
+import visuals.Draw;
+import visuals.ExecutionLog;
+
+import static visuals.DiagramView.tabPane;
+
+/**
+ * @author Pontus Laestadius
+ * @version 0.1
+ */
+class Decode {
+    String raw; // Raw string to be decoded.
+
+    /**
+     * @param string string to decode.
+     */
+    Decode(String string) {
+        raw = string;
+    }
+
+    /**
+     * Decodes the raw string and allocates it to it's associated diagram.
+     */
+    void execute() {
+        System.out.println(raw);
+
+        // If no string has been allocated, abort.
+        if (raw == null) return;
+
+        // Split the raw string in to fields.
+        int id_index = raw.indexOf(",");
+
+        String id_string = raw.substring(1, id_index);  // Retrieves the Diagram_ID.
+        int id = Integer.decode(id_string); // String to Integer.
+
+        if (raw.contains("simulation_finished")) { // If the simulation is finished.
+            write("Simulation finished"); // Write Simulation finished in the execution log.
+
+        } else if (raw.contains("print_information")) { // If it's a statement to only print.
+            write(raw); // Writes the raw data TODO replace with statement provided.
+            Net.push("{" + id + ", send_message}");
```

Comment:
If it is a print statement, yes. Do you want to change it?

Reply:
Im wondering since the send_message command will queue the next message to be simulated and that will cause the message queue on the client side to stack up messages